src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,240 @@
+                todoMessage(first, lst, store);
+                return false;
+            } else if (first.length() >= 8 && first.substring(0, 8).equals("deadline")) {
+                deadlineMessage(first, lst, store);
+                return false;
+            } else if (first.length() >= 5 && first.substring(0, 5).equals("event")) {
+                eventMessage(first, lst, store);
+                return false;
+            } else if (first.length() == 8 && first.substring(0, 6).equals("delete")) {
+                deleteMessage(first, lst, store);
+                return false;
+            } else if (first.length() >= 4 && first.substring(0, 4).equals("find")) {
+                findMessage(first, lst, store);
+                return false;
+            }
+            else {
+                throw new DukeException("I'm sorry, but I don't know what that means :-(");
+            }
+        } catch (DukeException ex) {
+            System.out.println(ex);
+            return false;
+        } catch (IOException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    /**
+     * Prints the list when user types "list"
+     *
+     * @param lst The tasklist
+     */
+    public static void listMessage(TaskList lst) throws DukeException{
+        if (lst.count() == 0) {
+            throw new DukeException("Your list is empty!!");
+        } else {
+            for (int i = 1; i < lst.count() + 1; i++) {
+                System.out.println((i) + ". " + lst.get(i - 1));
+            }
+        }
+    }
+
+    /**
+     * Marks a task as completed
+     *
+     * @param str The users input
+     * @param lst The tasklist
+     * @param storage the storage object
+     */
+    public static void markMessage(String str, TaskList lst, Storage storage) throws DukeException, IOException {
+        char index = str.charAt(5);
+        int number = Integer.parseInt(String.valueOf(index));
+        if(number > lst.count()) {
+            throw new DukeException("No such task.");
+        }
+        lst.markTask(number - 1);
+        storage.updateFile(lst);
+    }
+
+    /**
+     * Marks a task as incomplete
+     *
+     * @param str The users input
+     * @param lst The tasklist
+     * @param storage the storage object
+     */
+    public static void unmarkMessage(String str, TaskList lst, Storage storage) throws DukeException, IOException {
+        char index = str.charAt(7);
+        int number = Integer.parseInt(String.valueOf(index));
+        if(number > lst.count()) {
+            throw new DukeException("No such task.");
+        }
+        lst.unmarkTask(number - 1);
+        storage.updateFile(lst);
+    }
+
+    /**
+     * Adds a Todo to the tasklist
+     *
+     * @param str The users input
+     * @param lst The tasklist
+     * @param storage the storage object
+     */
+    public static void todoMessage(String str, TaskList lst, Storage storage) throws DukeException, IOException {
```

Comment:
Perhaps use verb for actions?
```suggestion
    public static void parseTodoMessage(String str, TaskList lst, Storage storage) throws DukeException, IOException {
```

Reply:
I found that many of the method names have the same issue.

Reply:
thanks for pointing that out! will change them :)